engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -386,7 +386,7 @@ namespace dmGameObject
                 params.m_Context = regist->m_ComponentTypes[i].m_Context;
                 params.m_ComponentIndex = i;
                 params.m_MaxComponentInstances = GetMaxComponentInstances(regist->m_ComponentTypes[i].m_NameHash, collection_desc);
-                params.m_MaxInstances = instances_in_collection;
+                params.m_MaxInstances = max_instances;
```

Comment:
and here we use max possible instances for "static" colletions it will be max_component counter, for dymaic the value max_instances from game.project